docs/adr/0006_component_constructors.md

```diff
@@ -0,0 +1,334 @@
+This schema is used to validate the component constructor YAML file and ensure that it adheres to the expected structure and data types.
+
+The idea is to provide a similar experience as the existing OCM CLI that focuses on
+
+- maintaining the format and structure of the `component-constructor.yaml` file schema
+- ensuring that input methods can be registered dynamically (via the plugin manager) or inbuilt (via the OCM CLI)
+- reusing the existing binding libraries to interact with OCM repositories
+
+## Decision Drivers
+    
+* Simplicity of Architecture: While component constructors are central to our user flow, they are also an integral
+  part of our development experience. As such, writing input methods must be simple and easy to adopt for even
+  less experienced engineers.
+* Extensibility: The component constructor schema must stay flexible enough to allow for the addition of new input methods
+  and the ability to extend existing ones.
+* Maintainability: The constructor logic and input methods should stay separated from existing OCI libraries and
+  binding libraries. This will allow us to maintain the component constructor schema and its input methods without
+  affecting the existing libraries.
+
+## Outcome
+
+TBD
+
+## A simplified `bindings/go/constructor` library
+
+The core idea of this proposal is the introduction of a new module `bindings/go/constructor` that will be responsible for
+
+* Providing centralized access and interaction with `component-constructor.yaml` style files and their schemas
+* Providing centralized extension points for adding new Input Methods for building component versions in addition
+  to directly referencing an access.
+
+## Input Methods and Contracts
+
+A key feature of the `component-constructor.yaml` file is the ability to define **inputs**—specifications that describe where to retrieve data and how to store it in a resource for a newly created component version.
+
+### Purpose of Input Methods
+
+An **input method** replaces the `access` field in a standard OCM resource or source. When specified, the input is interpreted at runtime during component construction and used to generate an appropriate access specification based on the processed input data.
+
+### Example
+
+The following `component-constructor.yaml` defines a simple input method:
+
+```yaml
+components:
+  - name: github.com/acme.org/helloworld
+    version: 1.0.0
+    provider:
+      name: internal
+    resources:
+      - name: testdata
+        type: blob
+        relation: local
+        input:
+          type: file
+          path: ./testdata/text.txt
+```
+
+In this example:
+
+- The input of type `file` tells the constructor to use the registered `file` input method.
+- This method processes `./testdata/text.txt` and generates an access specification.
+- The result is included in the final component descriptor.
+- The format of the generated access specification depends on the specific input method used.
```

Comment:
Perhaps show the resulting descriptor?

Reply:
I added a resulting descriptor example. please double check to make sure